src/asciidoctor-core-api.js

```diff
@@ -1486,7 +1486,8 @@ Document.prototype.getCompatMode = function () {
  * @memberof Document
  */
 Document.prototype.getSourcemap = function () {
-  return this.sourcemap
+  var sourcemap = this.sourcemap
+  return sourcemap === Opal.nil ? undefined : sourcemap
```

Comment:
I'm not sure about this one, maybe we should return false ? The sourcemap option is either true or false.

Reply:
Yes, this should be a boolean property.